tensorboard/webapp/metrics/store/metrics_reducers.ts

```diff
@@ -512,12 +513,23 @@ const reducer = createReducer(
         }
       }
 
-      const resolvedResult = buildOrReturnStateWithUnresolvedImportedPins(
-        state.unresolvedImportedPinnedCards,
-        newCardIds,
+      const {
+        nextCardToPinnedCopy,
+        nextPinnedCardToOriginal,
         nextCardMetadataMap,
+      } = updateCardMaps(
```

Comment:
Can you explain why we need to pass state.cardToPinnedCopy and state.pinnedCardToOriginal to this function, rather than just starting from scratch?

Reply:
Without passing these functions we don't have information to construct the maps. CardList contains a list of ids, `cardMetadataMap`, which is built from scratch in this change, maps `CardId` to card data (tag, runId etc). 
(due to the same reason we need to put back the pinned-original card id mapping to `cardMetadataMap`)

Reply:
I understand better now - I thought we could build these maps without looking at previous state but I realize now that isn't possible.